templates/simple_counter/web-leptos/src/main.rs

```diff
@@ -1,51 +1,41 @@
-use std::rc::Rc;
+mod core;
 
 use leptos::{
     component, create_effect, create_signal, view, IntoView, Scope, SignalGet, SignalUpdate,
 };
-use shared::{Capabilities, Core, Effect, Event};
+use {{core_name}}::Event;
 
 #[component]
 fn RootComponent(cx: Scope) -> impl IntoView {
-    let core = Rc::new(Core::new::<Capabilities>());
-    let (view_model, set_view_model) = create_signal(cx, core.view());
+    let core = core::new();
+    let (view, render) = create_signal(cx, core.view());
     let (event, set_event) = create_signal(cx, Event::Reset);
 
     create_effect(cx, move |_| {
-        let event = event.get();
-        let effects = core.process_event(event);
-        for effect in effects {
-            match effect {
-                Effect::Render(_) => {
-                    set_view_model.update(|view_model| *view_model = core.view());
-                }
-            };
-        }
+        core::update(&core, event.get(), render);
     });
 
     view! {cx,
-        <>
-            <section class="box container has-text-centered m-5">
-                <p class="is-size-5">{move || view_model.get().count}</p>
-                <div class="buttons section is-centered">
-                    <button class="button is-primary is-danger"
-                        on:click=move |_| set_event.update(|value| *value = Event::Reset)
-                    >
-                        {"Reset"}
-                    </button>
-                    <button class="button is-primary is-success"
-                        on:click=move |_| set_event.update(|value| *value = Event::Increment)
-                    >
-                        {"Increment"}
-                    </button>
-                    <button class="button is-primary is-warning"
-                        on:click=move |_| set_event.update(|value| *value = Event::Decrement)
-                    >
-                        {"Decrement"}
-                    </button>
-                </div>
-            </section>
-        </>
+        <section class="box container has-text-centered m-5">
+            <p class="is-size-5">{move || view.get().count}</p>
+            <div class="buttons section is-centered">
+                <button class="button is-primary is-danger"
+                    on:click=move |_| set_event.update(|value| *value = Event::Reset)
+                >
+                    {"Reset"}
+                </button>
+                <button class="button is-primary is-success"
+                    on:click=move |_| set_event.update(|value| *value = Event::Increment)
+                >
+                    {"Increment"}
+                </button>
+                <button class="button is-primary is-warning"
+                    on:click=move |_| set_event.update(|value| *value = Event::Decrement)
+                >
+                    {"Decrement"}
+                </button>
+            </div>
+        </section>
```

Comment:
I feel like having the templates do the counter is going too far. I'd gut any counter-y things out of these and rename the `templates/simple_counter` to `templates/shells`. 

It's goin to get really confusing when you point the doctor at your app and it starts recommending some UI for the counter example 😅

Reply:
Yeah I agree! Makes much more sense. Will do that

Reply:
Ok sure!